runtime/src/cost_tracker.rs

```diff
@@ -234,6 +264,42 @@ impl CostTracker {
             .saturating_sub(tx_cost.account_data_size);
         self.transaction_count = self.transaction_count.saturating_sub(1);
     }
+
+    /// Apply execution units adjustment to cost-tracker.
+    /// If adjustment causes arithmetic overflow, it bumps costs to their limits to
+    /// prevent adding further transactions into current block.
```

Comment:
This behavior feels strange to me. I would prefer if you saturating math instead. And to avoid the signed integer operations, I think you should replace this adjustment logic with a remove estimated cost operation and then an add actual cost operation

Reply:
> Yea, discussed saturating vs checked before. Basically, in case of overflow, it'd be saturated to u64::MAX, effectively same as what is it right now -- set block cost to/above limit, hence disallow new transactions. However in case of underflow, it'd be saturated to 0, it'd be unclear what to do with 0 then.

Ok regardless of saturing vs checked, I still think it'd be better avoid signed integer operations.

Reply:
tbh, signed integer operation doesn't bother me much, in fact, I feel it makes code more nature and cleaner. As delta of two number would be signed, and to accord that delta, one just do `add`. 

But the way it goes around to get the `i64` as delta is pretty cumbersome. Might as well to operate on `u64` as in this [commit](6e99eb1f0efc79629c826b343ac716ae9c92557f). See if you like this better, otherwise can always revert it.